app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageManager.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Infomaniak Mail - Android
+ * Copyright (C) 2023 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.ui.newMessage
+
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import com.infomaniak.mail.databinding.FragmentNewMessageBinding
+
+open class NewMessageManager {
+    private var _newMessageViewModel: NewMessageViewModel? = null
+    protected val newMessageViewModel: NewMessageViewModel get() = _newMessageViewModel!!
+    private var _binding: FragmentNewMessageBinding? = null
+    protected val binding: FragmentNewMessageBinding get() = _binding!!
+    private var _fragment: NewMessageFragment? = null
+    protected val fragment: NewMessageFragment get() = _fragment!!
+
+    val viewLifecycleOwner get() = fragment.viewLifecycleOwner
+    val childFragmentManager get() = fragment.childFragmentManager
+    val resources get() = fragment.resources
+    val context get() = fragment.requireContext()
```

Comment:
In fact I just realize that these manager should probably be FragmentScoped than ActivityScoped, as they were a part of NewMessageFragment.
It would greatly simplify all the fragment management

Reply:
In the end, passing the fragment would be more bothersome when we simply already have access to the fragment here, so we chose to keep this solution. The abstract class won't be instantiated so there's no need to make it injectable. Finally, I changed to the managers from ActivityScoped to FragmentScoped because here it doesn't change anything and it's more intuitive for these objects to be linked to the Fragment as they are some kind of extension of the fragment.